File: bootstrap/src/main/java/net/dontdrinkandroot/wicket/bootstrap/component/form/formgroup/FormGroupLocalDateTime.java
package net.dontdrinkandroot.wicket.bootstrap.component.form.formgroup;

import net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.InputGroup;
import net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.InputGroupLocalDateTime;
import net.dontdrinkandroot.wicket.component.form.LocalDateTimeTextField;
import org.apache.wicket.Component;
import org.apache.wicket.model.IModel;

import java.time.LocalDateTime;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class FormGroupLocalDateTime extends FormGroupInputGroup<LocalDateTime, LocalDateTime, LocalDateTimeTextField, InputGroupLocalDateTime>
{
    public FormGroupLocalDateTime(String id, IModel<String> labelModel, IModel<LocalDateTime> model)
    {
        super(id, labelModel, model);
    }

    @Override
    protected InputGroup<LocalDateTime, LocalDateTime, LocalDateTimeTextField> createInputGroup(String id)
    {
        return new InputGroupLocalDateTime(id, this.getModel())
        {
            @Override
            protected Component createInputGroupAddonBefore(String id)
            {
                return FormGroupLocalDateTime.this.createInputGroupAddonBefore(id);
            }

            @Override
            protected Component createInputGroupAddonAfter(String id)
            {
                return FormGroupLocalDateTime.this.createInputGroupAddonAfter(id);
            }
        };
    }
}

File: bootstrap/src/main/kotlin/net/dontdrinkandroot/wicket/bootstrap/component/form/formgroup/FormGroupLocalDateTime.kt
package net.dontdrinkandroot.wicket.bootstrap.component.form.formgroup

import net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.InputGroup
import net.dontdrinkandroot.wicket.bootstrap.component.form.inputgroup.InputGroupLocalDateTime
import net.dontdrinkandroot.wicket.component.form.LocalDateTimeTextField
import org.apache.wicket.Component
import org.apache.wicket.model.IModel
import java.time.LocalDateTime

class FormGroupLocalDateTime(id: String, model: IModel<LocalDateTime>, labelModel: IModel<String>) :
    FormGroupInputGroup<LocalDateTime, LocalDateTime, LocalDateTimeTextField, InputGroupLocalDateTime>(
        id,
        model,
        labelModel
    ) {

    override fun createInputGroup(id: String): InputGroup<LocalDateTime, LocalDateTime, LocalDateTimeTextField> {
        return object : InputGroupLocalDateTime(id, model) {
            override fun createAddonPrepend(id: String): Component {
                return this@FormGroupLocalDateTime.createInputGroupPrepend(id)
            }

            override fun createAddonAppend(id: String): Component {
                return this@FormGroupLocalDateTime.createInputGroupAppend(id)
            }
        }
    }
}